### PR TITLE
Re-style the "Move Columns" modal

### LINKED ISSUE #1667: Re-style the Move Columns modal
- Blocked by #1652

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #1667

## Workflow notes

**I'm out Wednesday and have unpredictable availability Thursday and Friday. I recommend pushing commits to this PR and merging instead of requesting changes.**

## Move to new table

| Figma | App |
| -- | -- |
| image | image |

| Figma | App |
| -- | -- |
| image | image |

## Move to existing linked table

| Figma | App |
| -- | -- |
| image | image |

## Related outstanding tickets

- The toast message will be styled as part of #2105.
- You might hit the backend bug #2138 if you test this locally. It does make it somewhat difficult to get the success toast message to appear.
- #2134 affects aesthetics with a narrow window, but I think it's pretty low priority.
- I opened #2139 for a small detail I hope we can defer to later.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@kgodey you might like to look over my "Figma vs App" screenshots above. There are some very minor wording changes.

**Comment by kgodey** [comment]
The changes seem fine @seancolsen

File: mathesar_ui/src/components/NameWithIcon.svelte [hunk 1]
```diff
@@ -9,12 +9,13 @@
   /** When true, the icon will be rendered within a box */
   export let iconHasBox = false;
   export let truncate = true;
+  export let bold = false;
```
Comment (rajatvijay) [inline mathesar_ui/src/components/NameWithIcon.svelte:12]: This seems a bit concerning for the future. We might end up having a lot of booleans for varying CSS props. No changes are needed urgently though.

**Review by rajatvijay — APPROVED** [review]
LGTM

**Pull request merged**